sdks/go/pkg/beam/core/graph/window/windows_test.go

```diff
@@ -0,0 +1,154 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package window
+
+import (
+	"testing"
+
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/core/graph/mtime"
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/core/typex"
+)
+
+func TestWindowEquality(t *testing.T) {
+	tests := []struct {
+		name        string
+		windowOne   typex.Window
+		windowTwo   typex.Window
+		expEquality bool
+	}{
+		{
+			"global window == global window",
+			GlobalWindow{},
+			GlobalWindow{},
+			true,
+		},
+		{
+			"interval window[0,10] == interval window[0,10]",
+			IntervalWindow{Start: 0, End: 10},
+			IntervalWindow{Start: 0, End: 10},
+			true,
+		},
+		{
+			"interval window[0,10] == interval window[11 20]",
+			IntervalWindow{Start: 0, End: 10},
+			IntervalWindow{Start: 11, End: 20},
+			false,
+		},
+		{
+			"interval window[0,10] == interval window[0,20]",
+			IntervalWindow{Start: 0, End: 10},
+			IntervalWindow{Start: 0, End: 20},
+			false,
+		},
+		{
+			"interval window[0,10] == interval window[5,10]",
+			IntervalWindow{Start: 0, End: 10},
+			IntervalWindow{Start: 5, End: 10},
+			false,
+		},
+		{
+			"global window == interval window[0,10]",
+			GlobalWindow{},
+			IntervalWindow{Start: 0, End: 10},
+			false,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			if got, want := test.windowOne.Equals(test.windowTwo), test.expEquality; got != want {
+				t.Errorf("got (%v).Equals(%v) == %v, want %v", test.windowOne, test.windowTwo, got, want)
```

Comment:
Small nit (here and elsewhere): the canonical/recommended pattern is function 1st, then got, then want - https://github.com/golang/go/wiki/TestComments#got-before-want - that's also what we use elsewhere AFIAK

Reply:
Good catch.